internal/bminventory/inventory_v2_handlers.go

```diff
@@ -672,6 +672,28 @@ func (b *bareMetalInventory) canServeHostIPXEScript(infraEnv *common.InfraEnv, m
 	return nil
 }
 
+func kernelArgsToSlice(infraEnv *common.InfraEnv) ([]string, error) {
```

Comment:
Are we going to need this elsewhere? Maybe this would be a good method to define on `common.InfraEnv` directly?

Marbe `infraEnv.UnmarshalKernelArgs()` or something.

Reply:
Not sure.  Currently this is the only place it is used.

Reply:
:man_shrugging: maybe just leave it as a part of `kernelArgsAppendStr` then and don't have it as a separate function at all.

Reply:
Changed.  Put all the functionality in this function.

Reply:
What do you think about going one more step and combining this with `kernelArgsAppendStr`?
`kernelArgsAppendStr` is really only one more call in addition to what's here.

Reply:
Seemed to me more logical to separate them because they are doing different things.
Many times I am for function separation.  It is probably a matter for style.
Anyway, I can change if you think it is essential.

Reply:
It's definitely not essential, but my heuristic for separating out a function is usually something like:

- Will the logic be reused in multiple places?
  - No, this function is only called in one place
- Is the logic complicated enough to need to be named to be understood?
  - No, the other function is a single conditional and an append
- Does the logic need its own unit-tests?
  - No, there are no direct unit tests for either of these
  
So in this case it seemed to me that one function would be better.
In fact, this is how I considered all of the functions I commented on in this PR.

But again, it's far from necessary. Clearly everything will work fine without making this change - it's just a suggestion along with some insight into my thought process.

Reply:
OK. So I prefer to leave it as is.